library/kani_core/src/arbitrary.rs

```diff
@@ -193,6 +193,87 @@ macro_rules! generate_arbitrary {
         pub mod slice {
             kani_core::slice_generator!();
         }
+
+        mod range_structures {
+            use super::{
+                Arbitrary,
+                core_path::{
+                    mem,
+                    ops::{Bound, Range, RangeFrom, RangeInclusive, RangeTo, RangeToInclusive},
+                },
+            };
+
+            impl<T> Arbitrary for Bound<T>
+            where
+                T: Arbitrary,
+            {
+                fn any() -> Self {
+                    match u8::any() % 3 {
+                        0 => Bound::Included(T::any()),
+                        1 => Bound::Excluded(T::any()),
+                        _ => Bound::Unbounded,
+                    }
+                }
+            }
+
+            impl<T> Arbitrary for Range<T>
+            where
+                T: Arbitrary + PartialOrd,
+            {
+                fn any() -> Self {
+                    let (mut first, mut second) = (T::any(), T::any());
+                    adjust(&mut first, &mut second);
```

Comment:
Technically, an adjustment shouldn't be done since `std::ops::Range { start: 10, end: 2 }` is a **valid** (although empty) range. The `Arbitrary` implementation for a type must cover the space of all valid values of the type for soundness purposes.

Reply:
`adjust` has been removed